pkgs/stdenv/generic/check-meta.nix

```diff
@@ -36,10 +36,10 @@ let
     attrs ? meta.license;
 
   hasWhitelistedLicense = assert areLicenseListsValid; attrs:
-    hasLicense attrs && builtins.elem attrs.meta.license whitelist;
+    hasLicense attrs && lib.lists.any (l: builtins.elem l whitelist) (lib.lists.toList attrs.meta.license);
 
   hasBlacklistedLicense = assert areLicenseListsValid; attrs:
-    hasLicense attrs && builtins.elem attrs.meta.license blacklist;
+    hasLicense attrs && lib.lists.any (l: builtins.elem l blacklist) (lib.lists.toList attrs.meta.license);
```

Comment:
It's debatable whether this should be `any` or `all` - if a package has multiple licenses and only one is on the blacklist I suppose it might be ok to use the non-blacklisted one, but I guess which one takes priority is probably a question for lawyers (https://en.wikipedia.org/wiki/Multi-licensing).  It felt safer to use `any` here.

The same logic could lead to using `all` for the whitelist above.  The logic here is more confusing though, since `hasDeniedUnfreeLicense` already supports lists and apparently will consider something unfree if *any* of the licenses is unfree (and providing it doesn't match `allowUnfreePredicate`).  So if you have a package with two licenses, one free and one unfree, and you want to use the whitelist to say that the unfree license is ok with you, you would have to add both the free and unfree license to the whitelist if we used `all`, even though the whitelist is only really relevant for unfree licenses.  Because of that I'm currently leaving it as using `any`, but open to suggestions.

Overall the asymmetry between the blacklist and the whitelist, and the ambiguity when multiple licenses are present makes this all quite confusing.  I tried to at least clarify the docs a little in https://github.com/NixOS/nixpkgs/pull/72075, but I think this could use a refactor.  It might be hard to clean it up properly without breaking existing configurations though.